Build error in std::bind

Building with clang 3.6, libc++ 1.0~svn199600-1 on Ubuntu leads to this error:

```
$ ./tools/unix/build_omim.sh
cd opening_hours/opening_hours_tests/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/johannes/src/omim/3party/opening_hours/opening_hours_tests/opening_hours_tests.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
In file included from /home/johannes/src/omim/drape/batcher.cpp:1:
In file included from /home/johannes/src/omim/drape/batcher.hpp:3:
In file included from /home/johannes/src/omim/drape/attribute_provider.hpp:3:
In file included from /home/johannes/src/omim/drape/pointers.hpp:3:
In file included from /home/johannes/src/omim/base/assert.hpp:3:
In file included from /home/johannes/src/omim/base/internal/message.hpp:2:
In file included from /home/johannes/src/omim/std/array.hpp:7:
In file included from /usr/include/c++/v1/array:104:
In file included from /usr/include/c++/v1/__tuple:16:
/usr/include/c++/v1/type_traits:433:83: error: no matching function for call to '__source'
    : public integral_constant<bool, sizeof(__libcpp_is_function_imp::__test<_Tp>(__libcpp_is_function_imp::__source<_Tp>())) == 1>
                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/v1/type_traits:438:14: note: in instantiation of template class 'std::__1::__libcpp_is_function<unsigned int () const,
      false>' requested here
    : public __libcpp_is_function<_Tp> {};
             ^
/usr/include/c++/v1/type_traits:443:97: note: in instantiation of template class 'std::__1::is_function<unsigned int () const>' requested
      here
template <class _Tp, class _Up> struct __libcpp_is_member_function_pointer<_Tp _Up::*> : public is_function<_Tp> {};
                                                                                                ^
/usr/include/c++/v1/type_traits:446:14: note: in instantiation of template class 'std::__1::__libcpp_is_member_function_pointer<unsigned
      int (dp::Batcher::CallbacksWrapper::*)() const>' requested here
    : public __libcpp_is_member_function_pointer<typename remove_cv<_Tp>::type> {};
             ^
/usr/include/c++/v1/type_traits:460:38: note: in instantiation of template class 'std::__1::is_member_function_pointer<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const>' requested here
                                    !is_member_function_pointer<_Tp>::value> {};
                                     ^
/usr/include/c++/v1/type_traits:3122:17: note: in instantiation of template class 'std::__1::is_member_object_pointer<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const>' requested here
                is_member_object_pointer<typename remove_reference<_Fp>::type>::value &&
                ^
/usr/include/c++/v1/type_traits:3129:1: note: (skipping 7 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
__invoke(_Fp&& __f, _A0&& __a0)
^
/usr/include/c++/v1/type_traits:3147:13: note: while substituting deduced template arguments into function template '__invoke' [with _Fp =
      std::__1::__bind<unsigned int (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *> &, _Args = <>]
            __invoke(_VSTD::declval<_Fp>(), _VSTD::declval<_Args>()...)
            ^
/usr/include/c++/v1/type_traits:3155:11: note: in instantiation of template class 'std::__1::__invokable_imp<std::__1::__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *> &>' requested here
          __invokable_imp<_Fp, _Args...>::value>
          ^
/usr/include/c++/v1/functional:1427:33: note: in instantiation of template class 'std::__1::__invokable<std::__1::__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *> &>' requested here
                                __invokable<_Fp&, _ArgTypes...>::value>
                                ^
/usr/include/c++/v1/functional:1454:41: note: in instantiation of default argument for '__callable<std::__1::__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *> >' required here
                                        __callable<_Fp>::value &&
                                        ^~~~~~~~~~~~~~~
/home/johannes/src/omim/drape/batcher.cpp:231:38: note: while substituting deduced template arguments into function template 'function'
      [with _Fp = std::__1::__bind<unsigned int (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>]
    callbacks.m_getAvailableVertex = bind(&CallbacksWrapper::GetAvailableVertexCount, &wrapper);
                                     ^
/usr/include/c++/v1/type_traits:424:28: note: candidate template ignored: substitution failure [with _Tp = unsigned int () const]:
      reference to function type 'unsigned int () const' cannot have 'const' qualifier
template <class _Tp> _Tp&  __source();
                     ~~~   ^
/home/johannes/src/omim/drape/batcher.cpp:231:36: error: no viable overloaded '='
    callbacks.m_getAvailableVertex = bind(&CallbacksWrapper::GetAvailableVertexCount, &wrapper);
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/v1/functional:1472:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'const std::__1::function<unsigned int ()>' for 1st
      argument
    function& operator=(const function&);
              ^
/usr/include/c++/v1/functional:1473:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'std::__1::function<unsigned int ()>' for 1st
      argument
    function& operator=(function&&) _NOEXCEPT;
              ^
/usr/include/c++/v1/functional:1474:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'nullptr_t' for 1st argument
    function& operator=(nullptr_t) _NOEXCEPT;
              ^
/usr/include/c++/v1/functional:1478:9: note: candidate template ignored: disabled by 'enable_if' [with _Fp = std::__1::__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>]
        __callable<typename decay<_Fp>::type>::value &&
        ^
/home/johannes/src/omim/drape/batcher.cpp:232:35: error: no viable overloaded '='
    callbacks.m_getAvailableIndex = bind(&CallbacksWrapper::GetAvailableIndexCount, &wrapper);
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/v1/functional:1472:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'const std::__1::function<unsigned int ()>' for 1st
      argument
    function& operator=(const function&);
              ^
/usr/include/c++/v1/functional:1473:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'std::__1::function<unsigned int ()>' for 1st
      argument
    function& operator=(function&&) _NOEXCEPT;
              ^
/usr/include/c++/v1/functional:1474:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'nullptr_t' for 1st argument
    function& operator=(nullptr_t) _NOEXCEPT;
              ^
/usr/include/c++/v1/functional:1478:9: note: candidate template ignored: disabled by 'enable_if' [with _Fp = std::__1::__bind<unsigned int
      (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>]
        __callable<typename decay<_Fp>::type>::value &&
        ^
make[2]: Entering directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_tests'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_tests'
cd opening_hours/opening_hours_integration_tests/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/johannes/src/omim/3party/opening_hours/opening_hours_integration_tests/opening_hours_integration_tests.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
3 errors generated.
make[1]: *** [../out/release/tmp/drape/batcher.o] Error 1
make[1]: Leaving directory `/home/johannes/src/omim-build-release/drape'
make: *** [sub-drape-make_first] Error 2
make: *** Waiting for unfinished jobs....
cd opening_hours/opening_hours_supported_features_tests/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/johannes/src/omim/3party/opening_hours/opening_hours_supported_features_tests/opening_hours_supported_features_tests.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
In file included from /home/johannes/src/omim/drape_frontend/tile_info.cpp:1:
In file included from /home/johannes/src/omim/drape_frontend/engine_context.hpp:3:
In file included from /home/johannes/src/omim/drape_frontend/map_shape.hpp:4:
In file included from /home/johannes/src/omim/drape_frontend/tile_key.hpp:3:
In file included from /home/johannes/src/omim/geometry/rect2d.hpp:3:
In file included from /home/johannes/src/omim/geometry/point2d.hpp:3:
In file included from /home/johannes/src/omim/base/assert.hpp:3:
In file included from /home/johannes/src/omim/base/internal/message.hpp:2:
In file included from /home/johannes/src/omim/std/array.hpp:7:
In file included from /usr/include/c++/v1/array:104:
In file included from /usr/include/c++/v1/__tuple:16:
/usr/include/c++/v1/type_traits:433:83: error: no matching function for call to '__source'
    : public integral_constant<bool, sizeof(__libcpp_is_function_imp::__test<_Tp>(__libcpp_is_function_imp::__source<_Tp>())) == 1>
                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/v1/type_traits:438:14: note: in instantiation of template class 'std::__1::__libcpp_is_function<bool () const, false>'
      requested here
    : public __libcpp_is_function<_Tp> {};
             ^
/usr/include/c++/v1/type_traits:443:97: note: in instantiation of template class 'std::__1::is_function<bool () const>' requested here
template <class _Tp, class _Up> struct __libcpp_is_member_function_pointer<_Tp _Up::*> : public is_function<_Tp> {};
                                                                                                ^
/usr/include/c++/v1/type_traits:446:14: note: in instantiation of template class 'std::__1::__libcpp_is_member_function_pointer<bool
      (df::TileInfo::*)() const>' requested here
    : public __libcpp_is_member_function_pointer<typename remove_cv<_Tp>::type> {};
             ^
/usr/include/c++/v1/type_traits:460:38: note: in instantiation of template class 'std::__1::is_member_function_pointer<bool
      (df::TileInfo::*)() const>' requested here
                                    !is_member_function_pointer<_Tp>::value> {};
                                     ^
/usr/include/c++/v1/type_traits:3122:17: note: in instantiation of template class 'std::__1::is_member_object_pointer<bool
      (df::TileInfo::*)() const>' requested here
                is_member_object_pointer<typename remove_reference<_Fp>::type>::value &&
                ^
/usr/include/c++/v1/type_traits:3129:1: note: (skipping 7 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
__invoke(_Fp&& __f, _A0&& __a0)
^
/usr/include/c++/v1/type_traits:3147:13: note: while substituting deduced template arguments into function template '__invoke' [with _Fp =
      std::__1::__bind<bool (df::TileInfo::*)() const, df::TileInfo *> &, _Args = <>]
            __invoke(_VSTD::declval<_Fp>(), _VSTD::declval<_Args>()...)
            ^
/usr/include/c++/v1/type_traits:3155:11: note: in instantiation of template class 'std::__1::__invokable_imp<std::__1::__bind<bool
      (df::TileInfo::*)() const, df::TileInfo *> &>' requested here
          __invokable_imp<_Fp, _Args...>::value>
          ^
/usr/include/c++/v1/functional:1427:33: note: in instantiation of template class 'std::__1::__invokable<std::__1::__bind<bool
      (df::TileInfo::*)() const, df::TileInfo *> &>' requested here
                                __invokable<_Fp&, _ArgTypes...>::value>
                                ^
/usr/include/c++/v1/functional:1454:41: note: in instantiation of default argument for '__callable<std::__1::__bind<bool
      (df::TileInfo::*)() const, df::TileInfo *> >' required here
                                        __callable<_Fp>::value &&
                                        ^~~~~~~~~~~~~~~
/home/johannes/src/omim/drape_frontend/tile_info.cpp:75:23: note: while substituting deduced template arguments into function template
      'function' [with _Fp = std::__1::__bind<bool (df::TileInfo::*)() const, df::TileInfo *>]
                      bind(&TileInfo::IsCancelled, this),
                      ^
/usr/include/c++/v1/type_traits:424:28: note: candidate template ignored: substitution failure [with _Tp = bool () const]: reference to
      function type 'bool () const' cannot have 'const' qualifier
template <class _Tp> _Tp&  __source();
                     ~~~   ^
/home/johannes/src/omim/drape_frontend/tile_info.cpp:74:16: error: no matching constructor for initialization of 'df::RuleDrawer'
    RuleDrawer drawer(bind(&TileInfo::InitStylist, this, _1 ,_2),
               ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/johannes/src/omim/drape_frontend/rule_drawer.hpp:31:3: note: candidate constructor not viable: no known conversion from '__bind<bool
      (df::TileInfo::*)() const, df::TileInfo *>' to 'const TCheckCancelledCallback' (aka 'const function<bool ()>') for 2nd argument
  RuleDrawer(TDrawerCallback const & drawerFn, TCheckCancelledCallback const & checkCancelled,
  ^
/home/johannes/src/omim/drape_frontend/rule_drawer.hpp:24:7: note: candidate constructor (the implicit copy constructor) not viable:
      requires 1 argument, but 5 were provided
class RuleDrawer
      ^
make[2]: Entering directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_integration_tests'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_integration_tests'
clang++ -c -pipe -stdlib=libc++ -fdata-sections -ffunction-sections -O3 -ffast-math -g -fPIC -std=c++11 -Wall -W -Wno-sign-compare -Wno-strict-aliasing -Wno-unused-parameter -Wno-unused-local-typedef -Wno-sign-conversion -Werror=return-type -D_REENTRANT -DRELEASE -D_RELEASE -DNDEBUG -DQT_NO_DEBUG -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-clang-libc++ -I/home/johannes/src/omim/drape_frontend -I/home/johannes/src/omim -I/home/johannes/src/omim/3party/boost -I/home/johannes/src/omim/3party/glm -I/home/johannes/src/omim/3party/protobuf/src -I/home/johannes/src/omim/3party/expat/lib -I/home/johannes/src/omim/3party/freetype/include -I. -o ../out/release/tmp/drape_frontend/tile_utils.o /home/johannes/src/omim/drape_frontend/tile_utils.cpp
2 errors generated.
make[1]: *** [../out/release/tmp/drape_frontend/tile_info.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[2]: Entering directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_supported_features_tests'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_supported_features_tests'
make[1]: Leaving directory `/home/johannes/src/omim-build-release/3party'
In file included from /home/johannes/src/omim/drape_frontend/drape_engine.cpp:1:
In file included from /home/johannes/src/omim/drape_frontend/drape_engine.hpp:3:
In file included from /home/johannes/src/omim/drape_frontend/backend_renderer.hpp:3:
In file included from /home/johannes/src/omim/drape_frontend/gui/layer_render.hpp:3:
In file included from /home/johannes/src/omim/drape_frontend/gui/skin.hpp:3:
In file included from /home/johannes/src/omim/drape/drape_global.hpp:3:
In file included from /home/johannes/src/omim/drape/color.hpp:4:
In file included from /home/johannes/src/omim/std/sstream.hpp:7:
In file included from /usr/include/c++/v1/sstream:174:
In file included from /usr/include/c++/v1/ostream:131:
In file included from /usr/include/c++/v1/ios:216:
In file included from /usr/include/c++/v1/__locale:15:
In file included from /usr/include/c++/v1/string:439:
In file included from /usr/include/c++/v1/algorithm:624:
/usr/include/c++/v1/type_traits:433:83: error: no matching function for call to '__source'
    : public integral_constant<bool, sizeof(__libcpp_is_function_imp::__test<_Tp>(__libcpp_is_function_imp::__source<_Tp>())) == 1>
                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/v1/type_traits:438:14: note: in instantiation of template class 'std::__1::__libcpp_is_function<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &) const, false>' requested here
    : public __libcpp_is_function<_Tp> {};
             ^
/usr/include/c++/v1/type_traits:443:97: note: in instantiation of template class 'std::__1::is_function<const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > (const std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> > &) const>' requested here
template <class _Tp, class _Up> struct __libcpp_is_member_function_pointer<_Tp _Up::*> : public is_function<_Tp> {};
                                                                                                ^
/usr/include/c++/v1/type_traits:446:14: note: in instantiation of template class 'std::__1::__libcpp_is_member_function_pointer<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const>' requested here
    : public __libcpp_is_member_function_pointer<typename remove_cv<_Tp>::type> {};
             ^
/usr/include/c++/v1/type_traits:3094:17: note: in instantiation of template class 'std::__1::is_member_function_pointer<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const>' requested here
                is_member_function_pointer<typename remove_reference<_Fp>::type>::value &&
                ^
/usr/include/c++/v1/type_traits:3101:1: note: in instantiation of default argument for '__invoke<const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *&, const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &>' required here
__invoke(_Fp&& __f, _A0&& __a0, _Args&& ...__args)
^~~~~~~~
/usr/include/c++/v1/type_traits:3147:13: note: (skipping 6 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
            __invoke(_VSTD::declval<_Fp>(), _VSTD::declval<_Args>()...)
            ^
/usr/include/c++/v1/type_traits:3147:13: note: while substituting deduced template arguments into function template '__invoke' [with _Fp =
      std::__1::__bind<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
      std::__1::placeholders::__ph<1> &> &, _Args = <const std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> > &>]
            __invoke(_VSTD::declval<_Fp>(), _VSTD::declval<_Args>()...)
            ^
/usr/include/c++/v1/type_traits:3155:11: note: in instantiation of template class 'std::__1::__invokable_imp<std::__1::__bind<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
      std::__1::placeholders::__ph<1> &> &, const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &>'
      requested here
          __invokable_imp<_Fp, _Args...>::value>
          ^
/usr/include/c++/v1/functional:1427:33: note: in instantiation of template class 'std::__1::__invokable<std::__1::__bind<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
      std::__1::placeholders::__ph<1> &> &, const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &>'
      requested here
                                __invokable<_Fp&, _ArgTypes...>::value>
                                ^
/usr/include/c++/v1/functional:1454:41: note: in instantiation of default argument for '__callable<std::__1::__bind<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
      std::__1::placeholders::__ph<1> &> >' required here
                                        __callable<_Fp>::value &&
                                        ^~~~~~~~~~~~~~~
/home/johannes/src/omim/drape_frontend/drape_engine.cpp:43:31: note: while substituting deduced template arguments into function template
      'function' [with _Fp = std::__1::__bind<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >
      (StringsBundle::*)(const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const,
      StringsBundle *, std::__1::placeholders::__ph<1> &>]
  guiSubsystem.SetLocalizator(bind(&StringsBundle::GetString, params.m_stringsBundle.get(), _1));
                              ^
/usr/include/c++/v1/type_traits:424:28: note: candidate template ignored: substitution failure [with _Tp = const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &) const]: reference to function type 'const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > (const std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> > &) const' cannot have 'const' qualifier
template <class _Tp> _Tp&  __source();
                     ~~~   ^
/home/johannes/src/omim/drape_frontend/drape_engine.cpp:43:31: error: no viable conversion from '__bind<const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *, std::__1::placeholders::__ph<1> &>' to
      'const TLocalizeStringFn' (aka 'const function<string (const string &)>')
  guiSubsystem.SetLocalizator(bind(&StringsBundle::GetString, params.m_stringsBundle.get(), _1));
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/v1/functional:1448:5: note: candidate constructor not viable: no known conversion from '__bind<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
      std::__1::placeholders::__ph<1> &>' to 'nullptr_t' for 1st argument
    function(nullptr_t) _NOEXCEPT : __f_(0) {}
    ^
/usr/include/c++/v1/functional:1449:5: note: candidate constructor not viable: no known conversion from '__bind<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
      std::__1::placeholders::__ph<1> &>' to 'const std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> > (const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &)> &' for
      1st argument
    function(const function&);
    ^
/usr/include/c++/v1/functional:1450:5: note: candidate constructor not viable: no known conversion from '__bind<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
      std::__1::placeholders::__ph<1> &>' to 'std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>,
      std::__1::allocator<char> > (const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &)> &&' for
      1st argument
    function(function&&) _NOEXCEPT;
    ^
/usr/include/c++/v1/functional:1454:41: note: candidate template ignored: disabled by 'enable_if' [with _Fp = std::__1::__bind<const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
      std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
      std::__1::placeholders::__ph<1> &>]
                                        __callable<_Fp>::value &&
                                        ^
/home/johannes/src/omim/drape_frontend/gui/drape_gui.hpp:36:49: note: passing argument to parameter 'fn' here
  void SetLocalizator(TLocalizeStringFn const & fn);
                                                ^
2 errors generated.
make[1]: *** [../out/release/tmp/drape_frontend/drape_engine.o] Error 1
make[1]: Leaving directory `/home/johannes/src/omim-build-release/drape_frontend'
make: *** [sub-drape_frontend-make_first] Error 2

```

It's a known clang 3.6.0 bug. Please try to upgrade it as described in docs.

> On 5.1.2016, at 04.56, Johannes Schlüter <notifications@github.com> wrote:
> 
> Building with clang 36, libc++ 10~svn199600-1 on Ubuntu leads to this error:
> 
> d opening_hours/opening_hours_tests/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/johannes/src/omim/3party/opening_hours/opening_hours_tests/opening_hours_testspro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> In file included from /home/johannes/src/omim/drape/batchercpp:1:
> In file included from /home/johannes/src/omim/drape/batcherhpp:3:
> In file included from /home/johannes/src/omim/drape/attribute_providerhpp:3:
> In file included from /home/johannes/src/omim/drape/pointershpp:3:
> In file included from /home/johannes/src/omim/base/asserthpp:3:
> In file included from /home/johannes/src/omim/base/internal/messagehpp:2:
> In file included from /home/johannes/src/omim/std/arrayhpp:7:
> In file included from /usr/include/c++/v1/array:104:
> In file included from /usr/include/c++/v1/__tuple:16:
> /usr/include/c++/v1/type_traits:433:83: error: no matching function for call to '__source'
>     : public integral_constant<bool, sizeof(__libcpp_is_function_imp::__test<_Tp>(__libcpp_is_function_imp::__source<_Tp>())) == 1>
>                                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/type_traits:438:14: note: in instantiation of template class 'std::__1::__libcpp_is_function<unsigned int () const,
>       false>' requested here
>     : public __libcpp_is_function<_Tp> {};
>              ^
> /usr/include/c++/v1/type_traits:443:97: note: in instantiation of template class 'std::__1::is_function<unsigned int () const>' requested
>       here
> template <class _Tp, class _Up> struct __libcpp_is_member_function_pointer<_Tp _Up::*> : public is_function<_Tp> {};
>                                                                                                 ^
> /usr/include/c++/v1/type_traits:446:14: note: in instantiation of template class 'std::__1::__libcpp_is_member_function_pointer<unsigned
>       int (dp::Batcher::CallbacksWrapper::*)() const>' requested here
>     : public __libcpp_is_member_function_pointer<typename remove_cv<_Tp>::type> {};
>              ^
> /usr/include/c++/v1/type_traits:460:38: note: in instantiation of template class 'std::__1::is_member_function_pointer<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const>' requested here
>                                     !is_member_function_pointer<_Tp>::value> {};
>                                      ^
> /usr/include/c++/v1/type_traits:3122:17: note: in instantiation of template class 'std::__1::is_member_object_pointer<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const>' requested here
>                 is_member_object_pointer<typename remove_reference<_Fp>::type>::value &&
>                 ^
> /usr/include/c++/v1/type_traits:3129:1: note: (skipping 7 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
> __invoke(_Fp&& __f, _A0&& __a0)
> ^
> /usr/include/c++/v1/type_traits:3147:13: note: while substituting deduced template arguments into function template '__invoke' [with _Fp =
>       std::__1::__bind<unsigned int (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *> &, _Args = <>]
>             __invoke(_VSTD::declval<_Fp>(), _VSTD::declval<_Args>())
>             ^
> /usr/include/c++/v1/type_traits:3155:11: note: in instantiation of template class 'std::__1::__invokable_imp<std::__1::__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *> &>' requested here
>           __invokable_imp<_Fp, _Args>::value>
>           ^
> /usr/include/c++/v1/functional:1427:33: note: in instantiation of template class 'std::__1::__invokable<std::__1::__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *> &>' requested here
>                                 __invokable<_Fp&, _ArgTypes>::value>
>                                 ^
> /usr/include/c++/v1/functional:1454:41: note: in instantiation of default argument for '__callable<std::__1::__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *> >' required here
>                                         __callable<_Fp>::value &&
>                                         ^~~~~~~~~~~~~~~
> /home/johannes/src/omim/drape/batchercpp:231:38: note: while substituting deduced template arguments into function template 'function'
>       [with _Fp = std::__1::__bind<unsigned int (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>]
>     callbacksm_getAvailableVertex = bind(&CallbacksWrapper::GetAvailableVertexCount, &wrapper);
>                                      ^
> /usr/include/c++/v1/type_traits:424:28: note: candidate template ignored: substitution failure [with _Tp = unsigned int () const]:
>       reference to function type 'unsigned int () const' cannot have 'const' qualifier
> template <class _Tp> _Tp&  __source();
>                      ~~~   ^
> /home/johannes/src/omim/drape/batchercpp:231:36: error: no viable overloaded '='
>     callbacksm_getAvailableVertex = bind(&CallbacksWrapper::GetAvailableVertexCount, &wrapper);
>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/functional:1472:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'const std::__1::function<unsigned int ()>' for 1st
>       argument
>     function& operator=(const function&);
>               ^
> /usr/include/c++/v1/functional:1473:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'std::__1::function<unsigned int ()>' for 1st
>       argument
>     function& operator=(function&&) _NOEXCEPT;
>               ^
> /usr/include/c++/v1/functional:1474:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'nullptr_t' for 1st argument
>     function& operator=(nullptr_t) _NOEXCEPT;
>               ^
> /usr/include/c++/v1/functional:1478:9: note: candidate template ignored: disabled by 'enable_if' [with _Fp = std::__1::__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>]
>         __callable<typename decay<_Fp>::type>::value &&
>         ^
> /home/johannes/src/omim/drape/batchercpp:232:35: error: no viable overloaded '='
>     callbacksm_getAvailableIndex = bind(&CallbacksWrapper::GetAvailableIndexCount, &wrapper);
>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/functional:1472:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'const std::__1::function<unsigned int ()>' for 1st
>       argument
>     function& operator=(const function&);
>               ^
> /usr/include/c++/v1/functional:1473:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'std::__1::function<unsigned int ()>' for 1st
>       argument
>     function& operator=(function&&) _NOEXCEPT;
>               ^
> /usr/include/c++/v1/functional:1474:15: note: candidate function not viable: no known conversion from '__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>' to 'nullptr_t' for 1st argument
>     function& operator=(nullptr_t) _NOEXCEPT;
>               ^
> /usr/include/c++/v1/functional:1478:9: note: candidate template ignored: disabled by 'enable_if' [with _Fp = std::__1::__bind<unsigned int
>       (dp::Batcher::CallbacksWrapper::*)() const, dp::Batcher::CallbacksWrapper *>]
>         __callable<typename decay<_Fp>::type>::value &&
>         ^
> make[2]: Entering directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_tests'
> make[2]: Nothing to be done for `first'
> make[2]: Leaving directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_tests'
> cd opening_hours/opening_hours_integration_tests/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/johannes/src/omim/3party/opening_hours/opening_hours_integration_tests/opening_hours_integration_testspro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> 3 errors generated
> make[1]: *** [/out/release/tmp/drape/batchero] Error 1
> make[1]: Leaving directory `/home/johannes/src/omim-build-release/drape'
> make: *** [sub-drape-make_first] Error 2
> make: *** Waiting for unfinished jobs
> cd opening_hours/opening_hours_supported_features_tests/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/johannes/src/omim/3party/opening_hours/opening_hours_supported_features_tests/opening_hours_supported_features_testspro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> In file included from /home/johannes/src/omim/drape_frontend/tile_infocpp:1:
> In file included from /home/johannes/src/omim/drape_frontend/engine_contexthpp:3:
> In file included from /home/johannes/src/omim/drape_frontend/map_shapehpp:4:
> In file included from /home/johannes/src/omim/drape_frontend/tile_keyhpp:3:
> In file included from /home/johannes/src/omim/geometry/rect2dhpp:3:
> In file included from /home/johannes/src/omim/geometry/point2dhpp:3:
> In file included from /home/johannes/src/omim/base/asserthpp:3:
> In file included from /home/johannes/src/omim/base/internal/messagehpp:2:
> In file included from /home/johannes/src/omim/std/arrayhpp:7:
> In file included from /usr/include/c++/v1/array:104:
> In file included from /usr/include/c++/v1/__tuple:16:
> /usr/include/c++/v1/type_traits:433:83: error: no matching function for call to '__source'
>     : public integral_constant<bool, sizeof(__libcpp_is_function_imp::__test<_Tp>(__libcpp_is_function_imp::__source<_Tp>())) == 1>
>                                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/type_traits:438:14: note: in instantiation of template class 'std::__1::__libcpp_is_function<bool () const, false>'
>       requested here
>     : public __libcpp_is_function<_Tp> {};
>              ^
> /usr/include/c++/v1/type_traits:443:97: note: in instantiation of template class 'std::__1::is_function<bool () const>' requested here
> template <class _Tp, class _Up> struct __libcpp_is_member_function_pointer<_Tp _Up::*> : public is_function<_Tp> {};
>                                                                                                 ^
> /usr/include/c++/v1/type_traits:446:14: note: in instantiation of template class 'std::__1::__libcpp_is_member_function_pointer<bool
>       (df::TileInfo::*)() const>' requested here
>     : public __libcpp_is_member_function_pointer<typename remove_cv<_Tp>::type> {};
>              ^
> /usr/include/c++/v1/type_traits:460:38: note: in instantiation of template class 'std::__1::is_member_function_pointer<bool
>       (df::TileInfo::*)() const>' requested here
>                                     !is_member_function_pointer<_Tp>::value> {};
>                                      ^
> /usr/include/c++/v1/type_traits:3122:17: note: in instantiation of template class 'std::__1::is_member_object_pointer<bool
>       (df::TileInfo::*)() const>' requested here
>                 is_member_object_pointer<typename remove_reference<_Fp>::type>::value &&
>                 ^
> /usr/include/c++/v1/type_traits:3129:1: note: (skipping 7 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
> __invoke(_Fp&& __f, _A0&& __a0)
> ^
> /usr/include/c++/v1/type_traits:3147:13: note: while substituting deduced template arguments into function template '__invoke' [with _Fp =
>       std::__1::__bind<bool (df::TileInfo::*)() const, df::TileInfo *> &, _Args = <>]
>             __invoke(_VSTD::declval<_Fp>(), _VSTD::declval<_Args>())
>             ^
> /usr/include/c++/v1/type_traits:3155:11: note: in instantiation of template class 'std::__1::__invokable_imp<std::__1::__bind<bool
>       (df::TileInfo::*)() const, df::TileInfo *> &>' requested here
>           __invokable_imp<_Fp, _Args>::value>
>           ^
> /usr/include/c++/v1/functional:1427:33: note: in instantiation of template class 'std::__1::__invokable<std::__1::__bind<bool
>       (df::TileInfo::*)() const, df::TileInfo *> &>' requested here
>                                 __invokable<_Fp&, _ArgTypes>::value>
>                                 ^
> /usr/include/c++/v1/functional:1454:41: note: in instantiation of default argument for '__callable<std::__1::__bind<bool
>       (df::TileInfo::*)() const, df::TileInfo *> >' required here
>                                         __callable<_Fp>::value &&
>                                         ^~~~~~~~~~~~~~~
> /home/johannes/src/omim/drape_frontend/tile_infocpp:75:23: note: while substituting deduced template arguments into function template
>       'function' [with _Fp = std::__1::__bind<bool (df::TileInfo::*)() const, df::TileInfo *>]
>                       bind(&TileInfo::IsCancelled, this),
>                       ^
> /usr/include/c++/v1/type_traits:424:28: note: candidate template ignored: substitution failure [with _Tp = bool () const]: reference to
>       function type 'bool () const' cannot have 'const' qualifier
> template <class _Tp> _Tp&  __source();
>                      ~~~   ^
> /home/johannes/src/omim/drape_frontend/tile_infocpp:74:16: error: no matching constructor for initialization of 'df::RuleDrawer'
>     RuleDrawer drawer(bind(&TileInfo::InitStylist, this, _1 ,_2),
>                ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /home/johannes/src/omim/drape_frontend/rule_drawerhpp:31:3: note: candidate constructor not viable: no known conversion from '__bind<bool
>       (df::TileInfo::*)() const, df::TileInfo *>' to 'const TCheckCancelledCallback' (aka 'const function<bool ()>') for 2nd argument
>   RuleDrawer(TDrawerCallback const & drawerFn, TCheckCancelledCallback const & checkCancelled,
>   ^
> /home/johannes/src/omim/drape_frontend/rule_drawerhpp:24:7: note: candidate constructor (the implicit copy constructor) not viable:
>       requires 1 argument, but 5 were provided
> class RuleDrawer
>       ^
> make[2]: Entering directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_integration_tests'
> make[2]: Nothing to be done for `first'
> make[2]: Leaving directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_integration_tests'
> clang++ -c -pipe -stdlib=libc++ -fdata-sections -ffunction-sections -O3 -ffast-math -g -fPIC -std=c++11 -Wall -W -Wno-sign-compare -Wno-strict-aliasing -Wno-unused-parameter -Wno-unused-local-typedef -Wno-sign-conversion -Werror=return-type -D_REENTRANT -DRELEASE -D_RELEASE -DNDEBUG -DQT_NO_DEBUG -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-clang-libc++ -I/home/johannes/src/omim/drape_frontend -I/home/johannes/src/omim -I/home/johannes/src/omim/3party/boost -I/home/johannes/src/omim/3party/glm -I/home/johannes/src/omim/3party/protobuf/src -I/home/johannes/src/omim/3party/expat/lib -I/home/johannes/src/omim/3party/freetype/include -I -o /out/release/tmp/drape_frontend/tile_utilso /home/johannes/src/omim/drape_frontend/tile_utilscpp
> 2 errors generated
> make[1]: *** [/out/release/tmp/drape_frontend/tile_infoo] Error 1
> make[1]: *** Waiting for unfinished jobs
> make[2]: Entering directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_supported_features_tests'
> make[2]: Nothing to be done for `first'
> make[2]: Leaving directory `/home/johannes/src/omim-build-release/3party/opening_hours/opening_hours_supported_features_tests'
> make[1]: Leaving directory `/home/johannes/src/omim-build-release/3party'
> In file included from /home/johannes/src/omim/drape_frontend/drape_enginecpp:1:
> In file included from /home/johannes/src/omim/drape_frontend/drape_enginehpp:3:
> In file included from /home/johannes/src/omim/drape_frontend/backend_rendererhpp:3:
> In file included from /home/johannes/src/omim/drape_frontend/gui/layer_renderhpp:3:
> In file included from /home/johannes/src/omim/drape_frontend/gui/skinhpp:3:
> In file included from /home/johannes/src/omim/drape/drape_globalhpp:3:
> In file included from /home/johannes/src/omim/drape/colorhpp:4:
> In file included from /home/johannes/src/omim/std/sstreamhpp:7:
> In file included from /usr/include/c++/v1/sstream:174:
> In file included from /usr/include/c++/v1/ostream:131:
> In file included from /usr/include/c++/v1/ios:216:
> In file included from /usr/include/c++/v1/__locale:15:
> In file included from /usr/include/c++/v1/string:439:
> In file included from /usr/include/c++/v1/algorithm:624:
> /usr/include/c++/v1/type_traits:433:83: error: no matching function for call to '__source'
>     : public integral_constant<bool, sizeof(__libcpp_is_function_imp::__test<_Tp>(__libcpp_is_function_imp::__source<_Tp>())) == 1>
>                                                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/type_traits:438:14: note: in instantiation of template class 'std::__1::__libcpp_is_function<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (const std::__1::basic_string<char,
>       std::__1::char_traits<char>, std::__1::allocator<char> > &) const, false>' requested here
>     : public __libcpp_is_function<_Tp> {};
>              ^
> /usr/include/c++/v1/type_traits:443:97: note: in instantiation of template class 'std::__1::is_function<const std::__1::basic_string<char,
>       std::__1::char_traits<char>, std::__1::allocator<char> > (const std::__1::basic_string<char, std::__1::char_traits<char>,
>       std::__1::allocator<char> > &) const>' requested here
> template <class _Tp, class _Up> struct __libcpp_is_member_function_pointer<_Tp _Up::*> : public is_function<_Tp> {};
>                                                                                                 ^
> /usr/include/c++/v1/type_traits:446:14: note: in instantiation of template class 'std::__1::__libcpp_is_member_function_pointer<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const>' requested here
>     : public __libcpp_is_member_function_pointer<typename remove_cv<_Tp>::type> {};
>              ^
> /usr/include/c++/v1/type_traits:3094:17: note: in instantiation of template class 'std::__1::is_member_function_pointer<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const>' requested here
>                 is_member_function_pointer<typename remove_reference<_Fp>::type>::value &&
>                 ^
> /usr/include/c++/v1/type_traits:3101:1: note: in instantiation of default argument for '__invoke<const std::__1::basic_string<char,
>       std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const std::__1::basic_string<char,
>       std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *&, const std::__1::basic_string<char,
>       std::__1::char_traits<char>, std::__1::allocator<char> > &>' required here
> __invoke(_Fp&& __f, _A0&& __a0, _Args&& __args)
> ^~~~~~~~
> /usr/include/c++/v1/type_traits:3147:13: note: (skipping 6 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
>             __invoke(_VSTD::declval<_Fp>(), _VSTD::declval<_Args>())
>             ^
> /usr/include/c++/v1/type_traits:3147:13: note: while substituting deduced template arguments into function template '__invoke' [with _Fp =
>       std::__1::__bind<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
>       std::__1::placeholders::__ph<1> &> &, _Args = <const std::__1::basic_string<char, std::__1::char_traits<char>,
>       std::__1::allocator<char> > &>]
>             __invoke(_VSTD::declval<_Fp>(), _VSTD::declval<_Args>())
>             ^
> /usr/include/c++/v1/type_traits:3155:11: note: in instantiation of template class 'std::__1::__invokable_imp<std::__1::__bind<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
>       std::__1::placeholders::__ph<1> &> &, const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &>'
>       requested here
>           __invokable_imp<_Fp, _Args>::value>
>           ^
> /usr/include/c++/v1/functional:1427:33: note: in instantiation of template class 'std::__1::__invokable<std::__1::__bind<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
>       std::__1::placeholders::__ph<1> &> &, const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &>'
>       requested here
>                                 __invokable<_Fp&, _ArgTypes>::value>
>                                 ^
> /usr/include/c++/v1/functional:1454:41: note: in instantiation of default argument for '__callable<std::__1::__bind<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
>       std::__1::placeholders::__ph<1> &> >' required here
>                                         __callable<_Fp>::value &&
>                                         ^~~~~~~~~~~~~~~
> /home/johannes/src/omim/drape_frontend/drape_enginecpp:43:31: note: while substituting deduced template arguments into function template
>       'function' [with _Fp = std::__1::__bind<const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >
>       (StringsBundle::*)(const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const,
>       StringsBundle *, std::__1::placeholders::__ph<1> &>]
>   guiSubsystemSetLocalizator(bind(&StringsBundle::GetString, paramsm_stringsBundleget(), _1));
>                               ^
> /usr/include/c++/v1/type_traits:424:28: note: candidate template ignored: substitution failure [with _Tp = const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (const std::__1::basic_string<char,
>       std::__1::char_traits<char>, std::__1::allocator<char> > &) const]: reference to function type 'const std::__1::basic_string<char,
>       std::__1::char_traits<char>, std::__1::allocator<char> > (const std::__1::basic_string<char, std::__1::char_traits<char>,
>       std::__1::allocator<char> > &) const' cannot have 'const' qualifier
> template <class _Tp> _Tp&  __source();
>                      ~~~   ^
> /home/johannes/src/omim/drape_frontend/drape_enginecpp:43:31: error: no viable conversion from '__bind<const std::__1::basic_string<char,
>       std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const std::__1::basic_string<char,
>       std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *, std::__1::placeholders::__ph<1> &>' to
>       'const TLocalizeStringFn' (aka 'const function<string (const string &)>')
>   guiSubsystemSetLocalizator(bind(&StringsBundle::GetString, paramsm_stringsBundleget(), _1));
>                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/functional:1448:5: note: candidate constructor not viable: no known conversion from '__bind<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
>       std::__1::placeholders::__ph<1> &>' to 'nullptr_t' for 1st argument
>     function(nullptr_t) _NOEXCEPT : __f_(0) {}
>     ^
> /usr/include/c++/v1/functional:1449:5: note: candidate constructor not viable: no known conversion from '__bind<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
>       std::__1::placeholders::__ph<1> &>' to 'const std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>,
>       std::__1::allocator<char> > (const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &)> &' for
>       1st argument
>     function(const function&);
>     ^
> /usr/include/c++/v1/functional:1450:5: note: candidate constructor not viable: no known conversion from '__bind<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
>       std::__1::placeholders::__ph<1> &>' to 'std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>,
>       std::__1::allocator<char> > (const std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &)> &&' for
>       1st argument
>     function(function&&) _NOEXCEPT;
>     ^
> /usr/include/c++/v1/functional:1454:41: note: candidate template ignored: disabled by 'enable_if' [with _Fp = std::__1::__bind<const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (StringsBundle::*)(const
>       std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > &) const, StringsBundle *,
>       std::__1::placeholders::__ph<1> &>]
>                                         __callable<_Fp>::value &&
>                                         ^
> /home/johannes/src/omim/drape_frontend/gui/drape_guihpp:36:49: note: passing argument to parameter 'fn' here
>   void SetLocalizator(TLocalizeStringFn const & fn);
>                                                 ^
> 2 errors generated
> make[1]: *** [/out/release/tmp/drape_frontend/drape_engineo] Error 1
> make[1]: Leaving directory `/home/johannes/src/omim-build-release/drape_frontend'
> make: *** [sub-drape_frontend-make_first] Error 2
> 
> —
> Reply to this email directly or view it on GitHub.
>

here it is:

macman@ubuntusvr:~/dev/omim$ tools/unix/build_omim.sh 
cd 3party/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/3party.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
cd base/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/base/base.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[1]: Entering directory `/home/macman/dev/omim-build-release/3party'
cd freetype/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/freetype/freetype.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/freetype'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/freetype'
cd fribidi/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/fribidi/fribidi.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/fribidi'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/fribidi'
cd minizip/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/minizip/minizip.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/minizip'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/minizip'
cd jansson/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/jansson/jansson.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[1]: Entering directory `/home/macman/dev/omim-build-release/base'
make[1]: Nothing to be done for `first'.
make[1]: Leaving directory `/home/macman/dev/omim-build-release/base'
cd coding/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/coding/coding.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/jansson'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/jansson'
cd tomcrypt/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/tomcrypt/tomcrypt.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/tomcrypt'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/tomcrypt'
cd protobuf/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/protobuf/protobuf.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/protobuf'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/protobuf'
cd osrm/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/osrm/osrm.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[1]: Entering directory `/home/macman/dev/omim-build-release/coding'
make[1]: Nothing to be done for `first'.
make[1]: Leaving directory `/home/macman/dev/omim-build-release/coding'
cd geometry/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/geometry/geometry.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[1]: Entering directory `/home/macman/dev/omim-build-release/geometry'
make[1]: Nothing to be done for `first'.
make[1]: Leaving directory `/home/macman/dev/omim-build-release/geometry'
cd editor/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/editor/editor.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/osrm'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/osrm'
cd expat/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/expat/expat.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/expat'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/expat'
cd succinct/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/succinct/succinct.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[1]: Entering directory `/home/macman/dev/omim-build-release/editor'
make[1]: Nothing to be done for `first'.
make[1]: Leaving directory `/home/macman/dev/omim-build-release/editor'
cd indexer/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/indexer/indexer.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/succinct'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/succinct'
cd pugixml/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/pugixml/pugixml.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/pugixml'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/pugixml'
cd liboauthcpp/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/liboauthcpp/liboauthcpp.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/liboauthcpp'
make[2]: Nothing to be done for `first'.
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/liboauthcpp'
cd opening_hours/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/opening_hours/opening_hours.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
make[2]: Entering directory `/home/macman/dev/omim-build-release/3party/opening_hours'
clang++ -c -pipe -stdlib=libc++ -fdata-sections -ffunction-sections -O3 -ffast-math -g -fPIC -std=c++11 -Wall -W -Wno-sign-compare -Wno-strict-aliasing -Wno-unused-parameter -Wno-unused-local-typedef -Wno-sign-conversion -Werror=return-type -D_REENTRANT -DRELEASE -D_RELEASE -DNDEBUG -DQT_NO_DEBUG -I/home/macman/dev/omim/3party/opening_hours -I. -I/home/macman/dev/omim -I/home/macman/dev/omim/3party/boost -I/home/macman/dev/omim/3party/glm -I/opt/qt55/mkspecs/linux-clang-libc++ -o ../../out/release/tmp/opening_hours/opening_hours.o /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp
make[1]: Entering directory `/home/macman/dev/omim-build-release/indexer'
make[1]: Nothing to be done for `first'.
make[1]: Leaving directory `/home/macman/dev/omim-build-release/indexer'
cd routing/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/routing/routing.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:25:
In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.hpp:27:
In file included from /usr/include/c++/v1/chrono:280:
/usr/include/c++/v1/type_traits:433:83: error: no matching function for call to '__source'
    : public integral_constant<bool, sizeof(__libcpp_is_function_imp::__test<_Tp>(__libcpp_is_function_imp::__source<_Tp>())) == 1>
                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/v1/type_traits:438:14: note: in instantiation of template class 'std::__1::__libcpp_is_function<bool () const, false>' requested here
    : public __libcpp_is_function<_Tp> {};
             ^
/usr/include/c++/v1/type_traits:443:97: note: in instantiation of template class 'std::__1::is_function<bool () const>' requested here
template <class _Tp, class _Up> struct __libcpp_is_member_function_pointer<_Tp _Up::*> : public is_function<_Tp> {};
                                                                                                ^
/usr/include/c++/v1/type_traits:446:14: note: in instantiation of template class 'std::__1::__libcpp_is_member_function_pointer<bool
      (osmoh::RuleSequence::*)() const>' requested here
    : public __libcpp_is_member_function_pointer<typename remove_cv<_Tp>::type> {};
             ^
/usr/include/c++/v1/type_traits:460:38: note: in instantiation of template class 'std::__1::is_member_function_pointer<bool (osmoh::RuleSequence::*)() const>'
      requested here
                                    !is_member_function_pointer<_Tp>::value> {};
                                     ^
/usr/include/c++/v1/type_traits:3122:17: note: in instantiation of template class 'std::__1::is_member_object_pointer<bool (osmoh::RuleSequence::*)() const>'
      requested here
                is_member_object_pointer<typename remove_reference<_Fp>::type>::value &&
                ^
/usr/include/c++/v1/__functional_base:399:1: note: in instantiation of default argument for '__invoke<bool (osmoh::RuleSequence::*)() const, const
      osmoh::RuleSequence &>' required here
__invoke(_Fp&& __f, _A0&& __a0)
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/v1/__functional_base:419:22: note: while substituting deduced template arguments into function template '__invoke' [with _Fp = bool
      (osmoh::RuleSequence::*)() const, _A0 = const osmoh::RuleSequence &, $2 = (no value)]
    typedef decltype(__invoke(_VSTD::declval<_Tp>(), _VSTD::declval<_Args>()...)) type;
                     ^
/usr/include/c++/v1/functional:1223:17: note: in instantiation of template class 'std::__1::__invoke_return<bool (osmoh::RuleSequence::*)() const, const
      osmoh::RuleSequence &>' requested here
       typename __invoke_return<type, _ArgTypes...>::type
                ^
/usr/include/c++/v1/algorithm:808:13: note: while substituting deduced template arguments into function template 'operator()'
      [with _ArgTypes = <const osmoh::RuleSequence &>]
        if (__pred(*__first))
            ^
/home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:787:15: note: in instantiation of function template specialization
      'std::__1::any_of<std::__1::__wrap_iter<const osmoh::RuleSequence *>, std::__1::__mem_fn<bool (osmoh::RuleSequence::*)() const> >' requested here
  return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
              ^
/usr/include/c++/v1/type_traits:424:28: note: candidate template ignored: substitution failure [with _Tp = bool () const]: reference to function type
      'bool () const' cannot have 'const' qualifier
template <class _Tp> _Tp&  __source();
                     ~~~   ^
In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:25:
In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.hpp:28:
In file included from /usr/include/c++/v1/iostream:38:
In file included from /usr/include/c++/v1/ios:216:
In file included from /usr/include/c++/v1/__locale:15:
In file included from /usr/include/c++/v1/string:439:
/usr/include/c++/v1/algorithm:808:13: error: attempt to use a deleted function
        if (__pred(*__first))
            ^
/home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:787:15: note: in instantiation of function template specialization
      'std::__1::any_of<std::__1::__wrap_iter<const osmoh::RuleSequence *>, std::__1::__mem_fn<bool (osmoh::RuleSequence::*)() const> >' requested here
  return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
              ^
/usr/include/c++/v1/type_traits:1001:5: note: '~__nat' has been explicitly marked deleted here
    ~__nat() = delete;
    ^
In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:25:
In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.hpp:28:
In file included from /usr/include/c++/v1/iostream:38:
In file included from /usr/include/c++/v1/ios:216:
In file included from /usr/include/c++/v1/__locale:18:
In file included from /usr/include/c++/v1/mutex:177:
/usr/include/c++/v1/functional:1226:22: error: attempt to use a deleted function
              return __invoke(__f_, _VSTD::forward<_ArgTypes>(__args)...);
                     ^
/usr/include/c++/v1/algorithm:808:13: note: in instantiation of function template specialization 'std::__1::__mem_fn<bool (osmoh::RuleSequence::*)()
      const>::operator()<const osmoh::RuleSequence &>' requested here
        if (__pred(*__first))
            ^
/home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:787:15: note: in instantiation of function template specialization
      'std::__1::any_of<std::__1::__wrap_iter<const osmoh::RuleSequence *>, std::__1::__mem_fn<bool (osmoh::RuleSequence::*)() const> >' requested here
  return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
              ^
/usr/include/c++/v1/type_traits:1001:5: note: '~__nat' has been explicitly marked deleted here
    ~__nat() = delete;
    ^
3 errors generated.
make[2]: *** [../../out/release/tmp/opening_hours/opening_hours.o] Error 1
make[2]: Leaving directory `/home/macman/dev/omim-build-release/3party/opening_hours'
make[1]: *** [sub-opening_hours-make_first] Error 2
make[1]: Leaving directory `/home/macman/dev/omim-build-release/3party'
make: *** [sub-3party-make_first] Error 2
make: *** Waiting for unfinished jobs....
make[1]: Entering directory `/home/macman/dev/omim-build-release/routing'
make[1]: Nothing to be done for `first'.
make[1]: Leaving directory `/home/macman/dev/omim-build-release/routing'

CC @mgsergio

> On 27.7.2016, at 06.16, antonyfr <notifications@github.com> wrote:
> 
> here it is:
> 
> macman@ubuntusvr:~/dev/omim$ tools/unix/build_omim.sh 
> cd 3party/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/3party.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> cd base/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/base/base.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory /home/macman/dev/omim-build-release/3party'
> cd freetype/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/freetype/freetype.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/freetype'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/freetype'
> cd fribidi/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/fribidi/fribidi.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/fribidi'
> make[2]: Nothing to be done forfirst'.
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/fribidi'
> cd minizip/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/minizip/minizip.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/minizip'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/minizip'
> cd jansson/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/jansson/jansson.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory /home/macman/dev/omim-build-release/base'
> make[1]: Nothing to be done forfirst'.
> make[1]: Leaving directory /home/macman/dev/omim-build-release/base'
> cd coding/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/coding/coding.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/jansson'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/jansson'
> cd tomcrypt/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/tomcrypt/tomcrypt.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/tomcrypt'
> make[2]: Nothing to be done forfirst'.
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/tomcrypt'
> cd protobuf/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/protobuf/protobuf.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/protobuf'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/protobuf'
> cd osrm/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/osrm/osrm.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory /home/macman/dev/omim-build-release/coding'
> make[1]: Nothing to be done forfirst'.
> make[1]: Leaving directory /home/macman/dev/omim-build-release/coding'
> cd geometry/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/geometry/geometry.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory/home/macman/dev/omim-build-release/geometry'
> make[1]: Nothing to be done for first'.
> make[1]: Leaving directory/home/macman/dev/omim-build-release/geometry'
> cd editor/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/editor/editor.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/osrm'
> make[2]: Nothing to be done forfirst'.
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/osrm'
> cd expat/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/expat/expat.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/expat'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/expat'
> cd succinct/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/succinct/succinct.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory /home/macman/dev/omim-build-release/editor'
> make[1]: Nothing to be done forfirst'.
> make[1]: Leaving directory /home/macman/dev/omim-build-release/editor'
> cd indexer/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/indexer/indexer.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/succinct'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/succinct'
> cd pugixml/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/pugixml/pugixml.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/pugixml'
> make[2]: Nothing to be done forfirst'.
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/pugixml'
> cd liboauthcpp/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/liboauthcpp/liboauthcpp.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/liboauthcpp'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/liboauthcpp'
> cd opening_hours/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/opening_hours/opening_hours.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/opening_hours'
> clang++ -c -pipe -stdlib=libc++ -fdata-sections -ffunction-sections -O3 -ffast-math -g -fPIC -std=c++11 -Wall -W -Wno-sign-compare -Wno-strict-aliasing -Wno-unused-parameter -Wno-unused-local-typedef -Wno-sign-conversion -Werror=return-type -D_REENTRANT -DRELEASE -D_RELEASE -DNDEBUG -DQT_NO_DEBUG -I/home/macman/dev/omim/3party/opening_hours -I. -I/home/macman/dev/omim -I/home/macman/dev/omim/3party/boost -I/home/macman/dev/omim/3party/glm -I/opt/qt55/mkspecs/linux-clang-libc++ -o ../../out/release/tmp/opening_hours/opening_hours.o /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp
> make[1]: Entering directory/home/macman/dev/omim-build-release/indexer'
> make[1]: Nothing to be done for first'.
> make[1]: Leaving directory/home/macman/dev/omim-build-release/indexer'
> cd routing/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/routing/routing.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:25:
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.hpp:27:
> In file included from /usr/include/c++/v1/chrono:280:
> /usr/include/c++/v1/type_traits:433:83: error: no matching function for call to '__source'
> : public integral_constant(__libcpp_is_function_imp::__source<Tp>())) == 1>
> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/type_traits:438:14: note: in instantiation of template class 'std::__1::__libcpp_is_function' requested here
> : public __libcpp_is_function<_Tp> {};
> ^
> /usr/include/c++/v1/type_traits:443:97: note: in instantiation of template class 'std::__1::is_function' requested here
> template struct __libcpp_is_member_function_pointer<_Tp _Up::> : public is_function<_Tp> {};
> ^
> /usr/include/c++/v1/type_traits:446:14: note: in instantiation of template class 'std::__1::__libcpp_is_member_function_pointer<bool
> (osmoh::RuleSequence::)() const>' requested here
> : public __libcpp_is_member_function_pointer::type> {};
> ^
> /usr/include/c++/v1/type_traits:460:38: note: in instantiation of template class 'std::__1::is_member_function_pointer'
> requested here
> !is_member_function_pointer<_Tp>::value> {};
> ^
> /usr/include/c++/v1/type_traits:3122:17: note: in instantiation of template class 'std::__1::is_member_object_pointer'
> requested here
> is_member_object_pointer::type>::value &&
> ^
> /usr/include/c++/v1/__functional_base:399:1: note: in instantiation of default argument for '__invoke osmoh::RuleSequence &>' required here
> __invoke(_Fp&& __f, _A0&& __a0)
> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/__functional_base:419:22: note: while substituting deduced template arguments into function template '__invoke' [with _Fp = bool
> (osmoh::RuleSequence::)() const, _A0 = const osmoh::RuleSequence &, $2 = (no value)]
> typedef decltype(__invoke(_VSTD::declval<_Tp>(), _VSTD::declval<_Args>()...)) type;
> ^
> /usr/include/c++/v1/functional:1223:17: note: in instantiation of template class 'std::__1::__invoke_return<bool (osmoh::RuleSequence::)() const, const
> osmoh::RuleSequence &>' requested here
> typename __invoke_return::type
> ^
> /usr/include/c++/v1/algorithm:808:13: note: while substituting deduced template arguments into function template 'operator()'
> [with _ArgTypes = ]
> if (__pred(__first))
> ^
> /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:787:15: note: in instantiation of function template specialization
> 'std::__1::any_of, std::__1::__mem_fn<bool (osmoh::RuleSequence::)() const> >' requested here
> return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
> ^
> /usr/include/c++/v1/type_traits:424:28: note: candidate template ignored: substitution failure [with _Tp = bool () const]: reference to function type
> 'bool () const' cannot have 'const' qualifier
> template _Tp& __source();
> ~~~ ^
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:25:
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.hpp:28:
> In file included from /usr/include/c++/v1/iostream:38:
> In file included from /usr/include/c++/v1/ios:216:
> In file included from /usr/include/c++/v1/__locale:15:
> In file included from /usr/include/c++/v1/string:439:
> /usr/include/c++/v1/algorithm:808:13: error: attempt to use a deleted function
> if (__pred(__first))
> ^
> /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:787:15: note: in instantiation of function template specialization
> 'std::__1::any_of, std::__1::__mem_fn<bool (osmoh::RuleSequence::)() const> >' requested here
> return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
> ^
> /usr/include/c++/v1/type_traits:1001:5: note: '~__nat' has been explicitly marked deleted here
> ~__nat() = delete;
> ^
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:25:
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.hpp:28:
> In file included from /usr/include/c++/v1/iostream:38:
> In file included from /usr/include/c++/v1/ios:216:
> In file included from /usr/include/c++/v1/__locale:18:
> In file included from /usr/include/c++/v1/mutex:177:
> /usr/include/c++/v1/functional:1226:22: error: attempt to use a deleted function
> return __invoke(__f, _VSTD::forward<_ArgTypes>(__args)...);
> ^
> /usr/include/c++/v1/algorithm:808:13: note: in instantiation of function template specialization 'std::__1::__mem_fn const>::operator()' requested here
> if (__pred(__first))
> ^
> /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:787:15: note: in instantiation of function template specialization
> 'std::__1::any_of, std::__1::__mem_fn<bool (osmoh::RuleSequence::)() const> >' requested here
> return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
> ^
> /usr/include/c++/v1/type_traits:1001:5: note: '~__nat' has been explicitly marked deleted here
> ~__nat() = delete;
> ^
> 3 errors generated.
> make[2]: *** [../../out/release/tmp/opening_hours/opening_hours.o] Error 1
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/opening_hours'
> make[1]: *** [sub-opening_hours-make_first] Error 2
> make[1]: Leaving directory/home/macman/dev/omim-build-release/3party'
> make: *** [sub-3party-make_first] Error 2
> make: *** Waiting for unfinished jobs....
> make[1]: Entering directory /home/macman/dev/omim-build-release/routing'
> make[1]: Nothing to be done forfirst'.
> make[1]: Leaving directory `/home/macman/dev/omim-build-release/routing'
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

We had the same issue once, It had something to do with spec files. But you use the right one (linux-clang-libc++).
We build mapsme on the system with the following properties:

Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.4 LTS
Release:	14.04
Codename:	trusty

rootuser@botswana:~$ clang —version
Ubuntu clang version 3.6.0-2ubuntu1~trusty1 (tags/RELEASE_360/final) (based on LLVM 3.6.0)
Target: x86_64-pc-linux-gnu
Thread model: posix
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8.4
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.9
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.9.3
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.8
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.8.4
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9.3
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/4.8
Candidate multilib: .;@m64
Selected multilib: .;@m64

rootuser@botswana:~$ /sbin/ldconfig -p | grep stdc++
	libstdc++.so.6 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libstdc++.so.6
	libstdc++.so.6 (libc6) => /usr/lib32/libstdc++.so.6

Do you have the same environment?

> On 27 Jul 2016, at 19:16, antonyfr <notifications@github.com> wrote:
> 
> here it is:
> 
> macman@ubuntusvr:~/dev/omim$ tools/unix/build_omim.sh 
> cd 3party/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/3party.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> cd base/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/base/base.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory /home/macman/dev/omim-build-release/3party'
> cd freetype/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/freetype/freetype.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/freetype'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/freetype'
> cd fribidi/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/fribidi/fribidi.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/fribidi'
> make[2]: Nothing to be done forfirst'.
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/fribidi'
> cd minizip/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/minizip/minizip.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/minizip'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/minizip'
> cd jansson/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/jansson/jansson.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory /home/macman/dev/omim-build-release/base'
> make[1]: Nothing to be done forfirst'.
> make[1]: Leaving directory /home/macman/dev/omim-build-release/base'
> cd coding/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/coding/coding.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/jansson'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/jansson'
> cd tomcrypt/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/tomcrypt/tomcrypt.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/tomcrypt'
> make[2]: Nothing to be done forfirst'.
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/tomcrypt'
> cd protobuf/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/protobuf/protobuf.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/protobuf'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/protobuf'
> cd osrm/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/osrm/osrm.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory /home/macman/dev/omim-build-release/coding'
> make[1]: Nothing to be done forfirst'.
> make[1]: Leaving directory /home/macman/dev/omim-build-release/coding'
> cd geometry/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/geometry/geometry.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory/home/macman/dev/omim-build-release/geometry'
> make[1]: Nothing to be done for first'.
> make[1]: Leaving directory/home/macman/dev/omim-build-release/geometry'
> cd editor/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/editor/editor.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/osrm'
> make[2]: Nothing to be done forfirst'.
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/osrm'
> cd expat/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/expat/expat.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/expat'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/expat'
> cd succinct/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/succinct/succinct.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[1]: Entering directory /home/macman/dev/omim-build-release/editor'
> make[1]: Nothing to be done forfirst'.
> make[1]: Leaving directory /home/macman/dev/omim-build-release/editor'
> cd indexer/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/indexer/indexer.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/succinct'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/succinct'
> cd pugixml/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/pugixml/pugixml.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/pugixml'
> make[2]: Nothing to be done forfirst'.
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/pugixml'
> cd liboauthcpp/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/liboauthcpp/liboauthcpp.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory/home/macman/dev/omim-build-release/3party/liboauthcpp'
> make[2]: Nothing to be done for first'.
> make[2]: Leaving directory/home/macman/dev/omim-build-release/3party/liboauthcpp'
> cd opening_hours/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/3party/opening_hours/opening_hours.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> make[2]: Entering directory /home/macman/dev/omim-build-release/3party/opening_hours'
> clang++ -c -pipe -stdlib=libc++ -fdata-sections -ffunction-sections -O3 -ffast-math -g -fPIC -std=c++11 -Wall -W -Wno-sign-compare -Wno-strict-aliasing -Wno-unused-parameter -Wno-unused-local-typedef -Wno-sign-conversion -Werror=return-type -D_REENTRANT -DRELEASE -D_RELEASE -DNDEBUG -DQT_NO_DEBUG -I/home/macman/dev/omim/3party/opening_hours -I. -I/home/macman/dev/omim -I/home/macman/dev/omim/3party/boost -I/home/macman/dev/omim/3party/glm -I/opt/qt55/mkspecs/linux-clang-libc++ -o ../../out/release/tmp/opening_hours/opening_hours.o /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp
> make[1]: Entering directory/home/macman/dev/omim-build-release/indexer'
> make[1]: Nothing to be done for first'.
> make[1]: Leaving directory/home/macman/dev/omim-build-release/indexer'
> cd routing/ && ( test -e Makefile || /usr/bin/qmake /home/macman/dev/omim/routing/routing.pro -spec linux-clang-libc++ CONFIG+=release -o Makefile ) && make -f Makefile 
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:25:
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.hpp:27:
> In file included from /usr/include/c++/v1/chrono:280:
> /usr/include/c++/v1/type_traits:433:83: error: no matching function for call to '__source'
> : public integral_constant(__libcpp_is_function_imp::__source<Tp>())) == 1>
> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/type_traits:438:14: note: in instantiation of template class 'std::__1::__libcpp_is_function' requested here
> : public __libcpp_is_function<_Tp> {};
> ^
> /usr/include/c++/v1/type_traits:443:97: note: in instantiation of template class 'std::__1::is_function' requested here
> template struct __libcpp_is_member_function_pointer<_Tp _Up::> : public is_function<_Tp> {};
> ^
> /usr/include/c++/v1/type_traits:446:14: note: in instantiation of template class 'std::__1::__libcpp_is_member_function_pointer<bool
> (osmoh::RuleSequence::)() const>' requested here
> : public __libcpp_is_member_function_pointer::type> {};
> ^
> /usr/include/c++/v1/type_traits:460:38: note: in instantiation of template class 'std::__1::is_member_function_pointer'
> requested here
> !is_member_function_pointer<_Tp>::value> {};
> ^
> /usr/include/c++/v1/type_traits:3122:17: note: in instantiation of template class 'std::__1::is_member_object_pointer'
> requested here
> is_member_object_pointer::type>::value &&
> ^
> /usr/include/c++/v1/__functional_base:399:1: note: in instantiation of default argument for '__invoke osmoh::RuleSequence &>' required here
> __invoke(_Fp&& __f, _A0&& __a0)
> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /usr/include/c++/v1/__functional_base:419:22: note: while substituting deduced template arguments into function template '__invoke' [with _Fp = bool
> (osmoh::RuleSequence::)() const, _A0 = const osmoh::RuleSequence &, $2 = (no value)]
> typedef decltype(__invoke(_VSTD::declval<_Tp>(), _VSTD::declval<_Args>()...)) type;
> ^
> /usr/include/c++/v1/functional:1223:17: note: in instantiation of template class 'std::__1::__invoke_return<bool (osmoh::RuleSequence::)() const, const
> osmoh::RuleSequence &>' requested here
> typename __invoke_return::type
> ^
> /usr/include/c++/v1/algorithm:808:13: note: while substituting deduced template arguments into function template 'operator()'
> [with _ArgTypes = ]
> if (__pred(__first))
> ^
> /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:787:15: note: in instantiation of function template specialization
> 'std::__1::any_of, std::__1::__mem_fn<bool (osmoh::RuleSequence::)() const> >' requested here
> return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
> ^
> /usr/include/c++/v1/type_traits:424:28: note: candidate template ignored: substitution failure [with _Tp = bool () const]: reference to function type
> 'bool () const' cannot have 'const' qualifier
> template _Tp& __source();
> ~~~ ^
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:25:
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.hpp:28:
> In file included from /usr/include/c++/v1/iostream:38:
> In file included from /usr/include/c++/v1/ios:216:
> In file included from /usr/include/c++/v1/__locale:15:
> In file included from /usr/include/c++/v1/string:439:
> /usr/include/c++/v1/algorithm:808:13: error: attempt to use a deleted function
> if (__pred(__first))
> ^
> /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:787:15: note: in instantiation of function template specialization
> 'std::__1::any_of, std::__1::__mem_fn<bool (osmoh::RuleSequence::)() const> >' requested here
> return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
> ^
> /usr/include/c++/v1/type_traits:1001:5: note: '~__nat' has been explicitly marked deleted here
> ~__nat() = delete;
> ^
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:25:
> In file included from /home/macman/dev/omim/3party/opening_hours/opening_hours.hpp:28:
> In file included from /usr/include/c++/v1/iostream:38:
> In file included from /usr/include/c++/v1/ios:216:
> In file included from /usr/include/c++/v1/__locale:18:
> In file included from /usr/include/c++/v1/mutex:177:
> /usr/include/c++/v1/functional:1226:22: error: attempt to use a deleted function
> return __invoke(__f, _VSTD::forward<_ArgTypes>(__args)...);
> ^
> /usr/include/c++/v1/algorithm:808:13: note: in instantiation of function template specialization 'std::__1::__mem_fn const>::operator()' requested here
> if (__pred(__first))
> ^
> /home/macman/dev/omim/3party/opening_hours/opening_hours.cpp:787:15: note: in instantiation of function template specialization
> 'std::__1::any_of, std::__1::__mem_fn<bool (osmoh::RuleSequence::)() const> >' requested here
> return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
> ^
> /usr/include/c++/v1/type_traits:1001:5: note: '~__nat' has been explicitly marked deleted here
> ~__nat() = delete;
> ^
> 3 errors generated.
> make[2]: *** [../../out/release/tmp/opening_hours/opening_hours.o] Error 1
> make[2]: Leaving directory /home/macman/dev/omim-build-release/3party/opening_hours'
> make[1]: *** [sub-opening_hours-make_first] Error 2
> make[1]: Leaving directory/home/macman/dev/omim-build-release/3party'
> make: *** [sub-3party-make_first] Error 2
> make: *** Waiting for unfinished jobs....
> make[1]: Entering directory /home/macman/dev/omim-build-release/routing'
> make[1]: Nothing to be done forfirst'.
> make[1]: Leaving directory `/home/macman/dev/omim-build-release/routing'
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/mapsme/omim/issues/1222#issuecomment-235637929>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AGWbq8vE1VR6xUAxmNgAz_Ko2q2sApxlks5qZ4R3gaJpZM4G-hEb>.
>

--- a/3party/opening_hours/opening_hours.cpp
+++ b/3party/opening_hours/opening_hours.cpp
@@ -784,21 +784,21 @@ bool OpeningHours::IsTwentyFourHours() const
 
 bool OpeningHours::HasWeekdaySelector() const
 {
-  return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
+  return std::any_of(m_rule.cbegin(), m_rule.cend(), std::mem_fn(&osmoh::RuleSequence::HasWeekdays));
 }
 
 bool OpeningHours::HasMonthSelector() const
 {
-  return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasMonths));
+  return std::any_of(m_rule.cbegin(), m_rule.cend(), std::mem_fn(&osmoh::RuleSequence::HasMonths));
 }
 
 bool OpeningHours::HasWeekSelector() const
 {
-  return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasWeeks));
+  return std::any_of(m_rule.cbegin(), m_rule.cend(), std::mem_fn(&osmoh::RuleSequence::HasWeeks));
 }
 
 bool OpeningHours::HasYearSelector() const
 {
-  return std::any_of(begin(m_rule), end(m_rule), std::mem_fn(&osmoh::RuleSequence::HasYears));
+  return std::any_of(m_rule.cbegin(), m_rule.cend(), std::mem_fn(&osmoh::RuleSequence::HasYears));
 }
 } // namespace osmoh

--- a/iphone/Maps/Classes/Common.h
+++ b/iphone/Maps/Classes/Common.h
@@ -20,7 +20,7 @@ static NSString * const kHaveAppleWatch = @"HaveAppleWatch";
 
 static CGFloat const kDefaultAnimationDuration = .2;
 
-static inline NSString * const kApplicationGroupIdentifier()
+static inline NSString * kApplicationGroupIdentifier()
 {
   static NSString * const productionGroupIdentifier = @"group.mapsme.watchkit.production";
   static NSString * const developerGroupIdentifier = @"group.mapsme.watchkit";

